torch/csrc/distributed/autograd/rpc_messages/rpc_with_profiling_req.cpp

```diff
@@ -0,0 +1,151 @@
+#include <torch/csrc/distributed/autograd/rpc_messages/rpc_with_profiling_req.h>
+#include <torch/csrc/distributed/rpc/utils.h>
+#include <torch/csrc/jit/serialization/pickle.h>
+#include <vector>
+
+namespace torch {
+namespace distributed {
+namespace autograd {
+
+using rpc::RpcCommandBase;
+
+// This constructor is called when creating the RpcWithProfilingReq on the
+// client.
+RpcWithProfilingReq::RpcWithProfilingReq(
+    rpc::worker_id_t fromWorkerId,
+    rpc::MessageType messageType,
+    rpc::Message&& wrappedMessage,
+    const torch::autograd::profiler::ProfilerConfig& profilerConfig)
+    : fromWorkerId_(fromWorkerId),
+      messageType_(messageType),
+      wrappedMessage_(std::move(wrappedMessage)),
+      profilerConfig_(profilerConfig) {
+  tensors_ = wrappedMessage_.tensors();
```

Comment:
This is constructing a copy. (not requesting changes in this PR) Does it make sense if we expose a `moveAll` method on `Message` that moves both `tensors` and `payload`?

Reply:
Yes, I think we could do something like this. This is an issue for `rpc_with_autograd.cpp` currently as well. Doesn't seem like we can move here as we need `wrappedMessage_` to be in a valid state in toMessageImpl.